demo/api/src/products/entities/raw-product.entity.ts

```diff
@@ -0,0 +1,116 @@
+import { Entity, Enum, ManyToOne, Property, Ref, types } from "@mikro-orm/core";
+import { CsvColumn } from "@src/importer/decorators/csv-column.decorator";
+import { TargetEntity } from "@src/importer/decorators/target-entity.decorator";
+import { BaseTargetEntity } from "@src/importer/entities/base-target.entity";
+import { Manufacturer } from "@src/products/entities/manufacturer.entity";
+import { IsArray, IsBoolean, IsDate, IsEnum, IsInt, IsOptional, IsString } from "class-validator";
+
+import { ProductDimensions, ProductDiscounts, ProductPriceRange, ProductStatus } from "./product.entity";
+import { ProductType } from "./product-type.enum";
+
+@Entity()
+@TargetEntity()
+export class RawProduct extends BaseTargetEntity {
```

Comment:
How does this entity combine with the "normal" product entity?

Reply:
This RawProduct holds the data from the dataSource more or less the way it's received.
Usually an additional checkin task can transform and or aggregate the data further. It would also setup relations. The import pipeline can be improved though, so it's more capable and might eliminate the need for a checkin in some cases.

Reply:
I removed rawProduct and now use the product entity